torch/_refs/__init__.py

```diff
@@ -2877,6 +2878,101 @@ def vsplit(
     return tensor_split(a, split_sizes, 0)
 
 
+# See maybe_wrap_dim in core
+def _maybe_wrap_dim(
+    dim: int,
+    dim_post_expr: int,
+    wrap_scalar: bool = True,
+) -> int:
+    # if dim_post_expr is 0 and wrap_scalar is true, then dim must be in the
+    # range [-1, 0]. This is a special case for scalar tensors and manifests in
+    # e.g. torch.sum(scalar_tensor, 0) Otherwise, dim should be in the range
+    # [-dim_post_expr, dim_post_expr-1].
+
+    # Fast path
+    if -dim_post_expr <= dim and dim < dim_post_expr:
+        # Branch-less version of dim + (dim < 0 ? dim_post_expr : 0)
+        return dim + dim_post_expr * (dim < 0)
+
+    # Check edge-cases out-of-line (wrapping scalars and out-of-bounds errors)
+    return _maybe_wrap_dim_slow(dim, dim_post_expr, wrap_scalar)
+
+
+def _maybe_wrap_dim_slow(
```

Comment:
no need to replicate aten structure here, it exists in aten for perf reasons, whereas refs are canonical implementations less concerned about perf, especially perf hit that would only happen once during tracing